### PR TITLE
docs: Output Format

### PR DESCRIPTION
**Pull request by rutujasurve94**
Docstring for `output` formatter

### CODE REVIEW

File: src/main/scala/com/raphtory/output/FileOutputFormat.scala [hunk 1]
```diff
@@ -7,6 +7,19 @@ import org.slf4j.LoggerFactory
...
+  *
+  * Usage while querying or running algorithmic tests:
+  *
+  * ```{code-block} scala
+  * import com.raphtory.output.FileOutputFormat
+  * algorithmTest(WeightedDegree(), outputFormat, 1, 32674, 10000, List(500, 1000, 10000))
+  * val testDir  =  "/tmp/raphtoryTest"
+  * val defaultOutputFormat: OutputFormat  =  FileOutputFormat(testDir)
+  * ```
+  */
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/FileOutputFormat.scala:22]: a ` ```{seealso}``` ` section would be nice pointing to the Raphtory Query API and the OutputFormat base class

File: src/main/scala/com/raphtory/core/algorithm/OutputFormat.scala [hunk 2]
```diff
@@ -3,6 +3,29 @@ package com.raphtory.core.algorithm
...
+  *
+  *        {s}`jobID: String`
+  *          : Job identifier for Raphtory job.
+  *
+  *        {s}`row: Row`
+  *          : Row for computation.
+  *
+  *        {s}`partitionID: Int``
+  *          : Paritition identifier.
+  */
```
[the remark this answers is not in the document]
Comment (rutujasurve94) [reply src/main/scala/com/raphtory/core/algorithm/OutputFormat.scala]: I've removed documentation for the abstract class for clarity 👍

File: src/main/scala/com/raphtory/output/FileOutputFormat.scala [hunk 3]
```diff
@@ -8,17 +8,41 @@ import org.slf4j.LoggerFactory
...
+  * def setSchema(): Schema[T]
+  *
+  * val schema: Schema[Array[Byte]] = Schema.BYTES
+  * val spout            = setSpout()
+  * val graphBuilder     = setGraphBuilder()
+  * val graph            = Raphtory.createTypedGraph[T](spout, graphBuilder, setSchema())
   * val testDir  =  "/tmp/raphtoryTest"
-  * val defaultOutputFormat: OutputFormat  =  FileOutputFormat(testDir)
+  * val outputFormat: FileOutputFormat  =  FileOutputFormat(testDir)
+  * graph.pointQuery(EdgeList(), outputFormat, 1595303181, List())
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/FileOutputFormat.scala:38]: Please implement the changes from the other pull request here as well, i.e., no need to have `setSpout` etc. functions, no schemas, `Raphtory.createTypedGraph` no longer exists...

File: src/main/scala/com/raphtory/output/FileOutputFormat.scala [hunk 4]
```diff
@@ -8,17 +8,41 @@ import org.slf4j.LoggerFactory
...
   * ```
+  *
+  *
+  *  ```{seealso}
+  *  [](com.raphtory.core.algorithm.OutputFormat),
+  *  [](com.raphtory.core.client.RaphtoryClient)
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/FileOutputFormat.scala:44]: I would add `[](com.raphtory.core.client.RaphtoryGraph)` and `[](com.raphtory.core.deploy.Raphtory)` as well

File: src/main/scala/com/raphtory/output/PulsarOutputFormat.scala [hunk 5]
```diff
@@ -7,24 +7,39 @@ import org.apache.pulsar.client.api.Producer
...
+  * def setSpout(): Spout[T]
+  * def setGraphBuilder(): GraphBuilder[T]
+  * def setSchema(): Schema[T]
+  *
+  * val schema: Schema[Array[Byte]] = Schema.BYTES
+  * val spout            = setSpout()
+  * val graphBuilder     = setGraphBuilder()
+  * val graph            = Raphtory.createTypedGraph[T](spout, graphBuilder, setSchema())
+  * val outputFormat: PulsarOutputFormat = PulsarOutputFormat("EdgeList")
+  * graph.pointQuery(EdgeList(), outputFormat, 1595303181, List())
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/PulsarOutputFormat.scala:35]: Same as for FileOutputFormat

File: src/main/scala/com/raphtory/output/PulsarOutputFormat.scala [hunk 6]
```diff
@@ -7,24 +7,39 @@ import org.apache.pulsar.client.api.Producer
 import org.apache.pulsar.client.api.Schema
 
 /**
-  * # PulsarOutputFormat
-  * `PulsarOutputFormat` writes output to a Raphtory Pulsar topic
+  * {s}`PulsarOutputFormat(pulsarTopic: String)`
+  *   : writes output output to a Raphtory Pulsar topic
+  *        {s}`pulsarTopic: String`
+  *          : Topic name for writing to Pulsar.
+  * Reference: [](com.raphtory.core.algorithm.OutputFormat)
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/PulsarOutputFormat.scala:14]: This is indented wrong and missing blank lines, does not format correctly when compiled. Can you please check the output after fixing this!

File: src/main/scala/com/raphtory/output/PulsarOutputFormat.scala [hunk 7]
```diff
@@ -7,24 +7,39 @@ import org.apache.pulsar.client.api.Producer
...
+  * val graphBuilder     = setGraphBuilder()
+  * val graph            = Raphtory.createTypedGraph[T](spout, graphBuilder, setSchema())
+  * val outputFormat: PulsarOutputFormat = PulsarOutputFormat("EdgeList")
+  * graph.pointQuery(EdgeList(), outputFormat, 1595303181, List())
   * ```
+  *
+  *
+  *  ```{seealso}
+  *  [](com.raphtory.core.algorithm.OutputFormat),
+  *  [](com.raphtory.core.client.RaphtoryClient.PointQuery)
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/PulsarOutputFormat.scala:41]: Second link does not exist, probably just copy from FileOutputFormat

File: src/main/scala/com/raphtory/output/FileOutputFormat.scala [hunk 8]
```diff
@@ -8,17 +8,41 @@ import org.slf4j.LoggerFactory
...
 
 /**
-  * # FileOutputFormat
-  * `FileOutputFormat` writes output for Raphtory Job and Partition for a pre-defined window and timestamp to File
+  * {s}`FileOutputFormat(filePath: String)`
+  *
+  *   : writes output for Raphtory Job and Partition for a pre-defined window and timestamp to File
+  *        {s}`filePath: String`
+  *          : Filepath for writing Raphtory output.
+  *   Reference: [](com.raphtory.core.algorithm.OutputFormat)
```
Comment (ljeub-pometry) [inline src/main/scala/com/raphtory/output/FileOutputFormat.scala:16]: This is not indented correctly and thus does not produce the expected output, please check!

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
This still has broken formatting, please fix and actually compile the documentation to check the output!

**Review by ljeub-pometry — APPROVED** [review]

**Pull request merged**